iss.py

```diff
@@ -1,10 +1,61 @@
 #!/usr/bin/env python
 
-__author__ = '???'
+__author__ = 'dbmiddle'
+
+import requests
+import time
+import turtle
 
 
 def main():
-    pass
+    # part_A:
+    res1 = requests.get('http://api.open-notify.org/astros.json')
+    res_obj = res1.json()
+    astros = res_obj['people']
+    for obj in astros:
+        print('{} is currently in space and onboard the {} spacecraft'.format(
+            obj['name'], obj['craft']))
+    print("There are {} total astronauts in space".format(len(astros)))
+
+    # part_B:
+    res2 = requests.get('http://api.open-notify.org/iss-now.json')
+    res_obj = res2.json()
+    print('On {}, the ISS was positioned at lat: {} lon: {}'.format(
+        time.ctime(res_obj['timestamp']), res_obj['iss_position']['latitude'],
+        res_obj['iss_position']['longitude']))
+
+    # part_C:
+    lat = res_obj['iss_position']['latitude']
+    lon = res_obj['iss_position']['longitude']
+    screen = turtle.Screen()
+    screen.setup(720, 360)
+    screen.setworldcoordinates(-180, -90, 180, 90)
+    screen.register_shape('iss.gif')
+    screen.bgpic('map.gif')
+    iss = turtle.Turtle()
+    iss.shape('iss.gif')
+    iss.penup()
+    iss.goto(float(lat), float(lon))
+
+    # part_D:
+    lat_indy = 39.7684
+    lon_indy = -86.1581
+    url = 'http://api.open-notify.org/iss-pass.json?lat=' + \
```

Comment:
Here you should not pass the url query parameters by composing your own string -- let the requests library do it for you (see [here](https://requests.readthedocs.io/en/master/user/quickstart/#passing-parameters-in-urls))  In this case your payload dict would be the pair of lat, lon values.